frameworks/kafka/tests/test_jmx.py

```diff
@@ -0,0 +1,204 @@
+    sdk_security.delete_secret("/test/integration/kafka/access")
+    sdk_security.delete_secret("/test/integration/kafka/keypass")
+
+
+@pytest.mark.sanity
+def test_secure_jmx_configuration():
+    foldered_name = sdk_utils.get_foldered_name(config.SERVICE_NAME)
+
+    try:
+        install_jmx_configured_kafka()
+        broker_task_id_0 = sdk_tasks.get_task_ids(foldered_name)[0]
+        install_jmxterm(task_id=broker_task_id_0)
+        generate_jmx_command_files(task_id=broker_task_id_0)
+
+        cmd = "export JAVA_HOME=$(ls -d ${MESOS_SANDBOX}/jdk*/jre) && " \
+              "${JAVA_HOME}/bin/java " \
+              "-Duser.home=${MESOS_SANDBOX} " \
+              "-Djdk.tls.client.protocols=TLSv1.2 -Djavax.net.ssl.trustStore=${MESOS_SANDBOX}/jmx/trust_store " \
+              "-Djavax.net.ssl.trustStorePassword=deleteme " \
+              "-Djavax.net.ssl.keyStore=${MESOS_SANDBOX}/jmx/key_store -Djavax.net.ssl.keyStorePassword=deleteme " \
+              "-Djavax.net.ssl.trustStoreType=JKS -Djavax.net.ssl.keyStoreType=JKS -jar jmxterm-1.0.1-uber.jar " \
+              "-l service:jmx:rmi:///jndi/rmi://${MESOS_CONTAINER_IP}:31199/jmxrmi -u admin -p adminpassword " \
+              "-s -v silent -n < jmx_beans_command.txt"
+
+        full_cmd = "bash -c '{}'".format(cmd)
+
+        _, output, _ = sdk_cmd.run_cli("task exec {} {}".format(broker_task_id_0, full_cmd), print_output=True)
+
+        assert "kafka.server:type=kafka-metrics-count" in output
+        assert "kafka.server:name=BrokerState,type=KafkaServer" in output
+
+        cmd = "export JAVA_HOME=$(ls -d ${MESOS_SANDBOX}/jdk*/jre) && " \
+              "${JAVA_HOME}/bin/java " \
+              "-Duser.home=${MESOS_SANDBOX} " \
+              "-Djdk.tls.client.protocols=TLSv1.2 -Djavax.net.ssl.trustStore=${MESOS_SANDBOX}/jmx/trust_store " \
+              "-Djavax.net.ssl.trustStorePassword=deleteme " \
+              "-Djavax.net.ssl.keyStore=${MESOS_SANDBOX}/jmx/key_store -Djavax.net.ssl.keyStorePassword=deleteme " \
+              "-Djavax.net.ssl.trustStoreType=JKS -Djavax.net.ssl.keyStoreType=JKS -jar jmxterm-1.0.1-uber.jar " \
+              "-l service:jmx:rmi:///jndi/rmi://${MESOS_CONTAINER_IP}:31199/jmxrmi -u admin -p adminpassword " \
+              "-s -v silent -n < jmx_domains_command.txt"
+
+        full_cmd = "bash -c '{}'".format(cmd)
+
+        rc, output, stderr = sdk_cmd.run_cli("task exec {} {}".format(broker_task_id_0, full_cmd), print_output=True)
+        print(rc)
+        print(output)
+        print(stderr)
+
+        assert "kafka.server" in output
+        assert "kafka.controller" in output
+
+    finally:
+        sdk_install.uninstall(config.PACKAGE_NAME, foldered_name)
+        uninstall_jmx_secrets()
+
+
+@pytest.mark.sanity
+def test_secure_jmx_dcos_crt():
+    foldered_name = sdk_utils.get_foldered_name(config.SERVICE_NAME)
+
+    try:
+        install_jmx_configured_kafka(self_signed_trust_store=False)
```

Comment:
There is a way to parameterize tests [like this](https://github.com/mesosphere/spark-build/blob/master/tests/test_spark_metrics.py#L25) so you could pass `self_signed_trust_store=False|True` and then depending on the param choose which `cmd` to run when checking `beans_command`. Not a hard ask, just throwing it out there in case you wanted to do it.

Reply:
thats awesome!